### DISCUSSION TITLE
How do I navigate to the signin-page when there is no session ?

### DISCUSSION THREAD

**Discussion by Cliff-R-K** [nextauthjs/next-auth#2471] [Help]
### Question 💬

How do I automaticly navigate to the /loginpage when no session is found (without clicking a button) ?

I have tried adding both `signIn()` and `router.push('/api/auth/signin')` after `if(session)` but none of the methods work from me.

```
import React from "react";
import { useSession, signIn, signOut } from "next-auth/client";
import { useRouter } from 'next/router'
export default function Home() {

  const [session, loading] = useSession();
  const router = useRouter()

  if (session) {
    console.log("session = true")
    router.push('/blogs')
    return (
      <>
        Signed in as {session.user.name} <br />
        <button onClick={() => signOut()}>Sign out</button>
      </>
    );
  }
  console.log("session = false")
 
  return (
    <>
      Not signed in <br />
      <button onClick={() => signIn()}>Sign in</button>
    </> 
  );
}
```

### How to reproduce ☕️

https://codesandbox.io/s/angry-chandrasekhar-zzlnu

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
You probably want to check out `useEffect`: https://reactjs.org/docs/hooks-effect.html
